### PR TITLE
Add osx-64

### LINKED ISSUE #1: OSX Builds
The initial pull request didn't handle OSX, mostly likely due to some upstream use of overly-modern calls. Let's get it working.

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
  * [x] just want to _add_ osx, not push new win/linux builds, hopefully 
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

Notes

- fixes #1

### CODE REVIEW

**Comment by bollwyvl** [comment]
Looking into the [knowledge base](https://conda-forge.org/docs/maintainer/knowledge_base.html?highlight=newer%20osx#requiring-newer-macos-sdks) it looks like updating the deployment target it The Way. @conda-forge/core what are the concerns about this? No rush! If I can't get  something that _works_ up, it's moot anyway.

**Comment by bollwyvl** [comment]
@conda-forge-admin please restart ci

**Comment by bollwyvl** [comment]
@conda-forge-admin please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -100,7 +100,7 @@ source:
 
 build:
   number: 0
-  skip: true  # [osx]
+  skip: true
```
Comment (chrisburr) [inline recipe/meta.yaml]:
```suggestion
```

**Comment by chrisburr** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (bollwyvl) [reply recipe/meta.yaml]: oh man, it's too early to be fighting this beast, perhaps.

**Comment by chrisburr** [comment]
For now it seems like the failure is that it's looking for OpenSSL in the wrong place: `ld: warning: directory not found for option '-L/usr/local/opt/openssl@1.1/lib'`

**Comment by bollwyvl** [comment]
> `ld: warning: directory not found for option '-L/usr/local/opt/openssl@1.1/lib'`

I think they were just that, warnings... the `CFLAGS` which are getting handed down via to `GOCFLAGS` already included an `-i` pointing at `PREFIX`... but obviously a bit out of my depth :grimacing:

**Comment by bollwyvl** [comment]
Hm... it's hanging:
```
WARNING :: Failed to get_static_lib_exports(/Users/runner/miniforge3/conda-bld/go-ipfs_1615664522802/_build_env/go/src/go/internal/gccgoimporter/testdata/libimportsar.a)
WARNING :: Failed to get_static_lib_exports(/Users/runner/miniforge3/conda-bld/go-ipfs_1615664522802/_build_env/lib/clang/11.1.0/lib/libclang_rt.cc_kext_i386_osx.a)
WARNING :: Failed to get_static_lib_exports(/Users/runner/miniforge3/conda-bld/go-ipfs_1615664522802/_build_env/lib/clang/11.1.0/lib/libclang_rt.builtins_i386_osx.a)
   INFO: sysroot: '/Applications/Xcode_12.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk/' files: '['usr/share/man/mann/zero.n', 'usr/share/man/mann/yencode.n', 'usr/share/man/mann/yaml.n', 'usr/share/man/mann/xsxp.n']'
```

**Comment by bollwyvl** [comment]
Welp, took 55 minutes, but it finished...

**Pull request merged**